### DISCUSSION TITLE
HxGrid Adding Items to a grid with InLineEditing.

### DISCUSSION THREAD

**Discussion by bluisanajmr** [havit/Havit.Blazor#775] [Q&A]
Hello, 

I have a grid with InlineEditing enabled. 

`<HxGrid MultiSelectionEnabled="true"
        @ref="InvoiceLineItemDataGrid" TItem="InvoiceLineItemGrid" Responsive="true" DataProvider="GetGridData" PageSize="10" 
        SelectionEnabled="true"
        @bind-SelectedDataItem="edititem"
        @bind-SelectedDataItems="MultiInvoiceLineItems" FooterRowCssClass="footer-row" TableCssClass="footerborder">`

I have a button on the same page with this grid that allows the user to add items to the grid.  This is done by adding a blank item to the datasource that the grid is using and then calling the RefreshDataAsync on the grid to display the new item.  

//refresh to bring up the added item
await InvoiceLineItemDataGrid.RefreshDataAsync();

This works but I would like newly added items to be in inline editing mode in the grid by default.  I would also like to be able to navigate to the page that the data item gets added to (the last page).  How can I accomplish this?

**Comment by hakenr** [comment]
> This works but I would like newly added items to be in inline editing mode in the grid by default

Set `edititem = newlyAddedItem`;

> I would also like to be able to navigate to the page that the data item gets added to (the last page). 

This is achievable by binding to `@bind-CurrentUserState="myUserState` and setting the `myUserState.PageIndex` property.

**Comment by bluisanajmr** [comment]
Thanks Hakenr, 

I almost have this almost working. My problem is that setting the griduserstate below doesn't cause the RefreshdataAsync() on the grid to fire until after my method has executed.  I also don't see a way to await instantiating the griduserstate.  The reason this is a problem is that the fulldatalist and returnlist are built from API calls inside of the RefreshdataAsync.  This makes the returnlist have the paged data for what ever is in the grid before I set the griduserstate to the last page.  

`

```
//refresh to bring up the added item
await InvoiceLineItemDataGrid.RefreshDataAsync();


//make sure we have items in our grid data list
if (fulldatalist.Count > 0)
{
    //get current pagesize of the grid
    int? gridpagesize = InvoiceLineItemDataGrid.PageSize;
    int numpages=0;

    if (gridpagesize != null && gridpagesize.HasValue)
    {
        //get the number of pages in our grid with current data list
        numpages = (fulldatalist.Count / gridpagesize.Value);
    }

    //make sure we are on the last page of data in the grid.
    gridUserState = new GridUserState(numpages, null);

    //put the added line item into edit mode.
    //we are always adding our lineitems to the end of the list 
    edititem = returnlist.Last();
}
else
{
    //put the added line item into edit mode.
    //we are always adding our lineitems to the end of the list
    edititem = returnlist.Last();
}
```
`
